### PR TITLE
refactor: [TECH-920][TECH-660] use ScopeSelector in the New page

### PR DESCRIPTION
**Pull request by simonadomnisoru**

### CODE REVIEW

File: src/core_modules/capture-core/components/Pages/New/TopBar.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            customActionsOnProgramIdReset={[
+                cleanUpDataEntry(NEW_TEI_DATA_ENTRY_ID),
+                cleanUpDataEntry(NEW_SINGLE_EVENT_DATA_ENTRY_ID),
+                cleanUpDataEntry(NEW_RELATIONSHIP_EVENT_DATA_ENTRY_ID),
+            ]}
+            customActionsOnOrgUnitIdReset={[
+                cleanUpDataEntry(NEW_TEI_DATA_ENTRY_ID),
+                cleanUpDataEntry(NEW_SINGLE_EVENT_DATA_ENTRY_ID),
+                cleanUpDataEntry(NEW_RELATIONSHIP_EVENT_DATA_ENTRY_ID),
+            ]}
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/Pages/New/TopBar.js:78]: We should strive to get rid of these properties and the functionality for them inside the scope selector (if we need some actions to be dispatched we can do it in `onResetOrgUnitId` or `onResetProgramId`). Been doing some testing today and It seems these clean ups are also being run from elsewhere (RegistrationDataEntry.container), so I don't think we need to dispatch these here at all. Am I mistaken?

**Review by JoakimSM — COMMENTED** [review]
Looks like a pretty good first step to get out of the selector / syncing problem. I will push a PR that deals with a problem I found (deselecting org. unit also deselects category options) and does some minor clean up in an epic (an epic we should prioritise to do a proper refactor of soon, I will create a ticket). 

Great if you have a look at the other comment I made.

File: src/core_modules/capture-core/components/Pages/New/TopBar.js (hunk 1 above)
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/Pages/New/TopBar.js]: Hey, 
You were right those cleanups were already run from inside `RegistrationDataEntry.container`. I clean the code of them inside ScopeSelector in the latest commit. 
Thank you!

**Review by JoakimSM — APPROVED** [review]
🎉

**Pull request merged**